src/extensions/nuget/Microsoft.DotNet.UpgradeAssistant.Extensions.NuGet/NuGetTransitiveDependencyIdentifier.cs

```diff
@@ -98,28 +116,24 @@ public async Task<TransitiveClosureCollection> GetTransitiveDependenciesAsync(IE
             dependencyGraphSpec.AddProject(spec);
             dependencyGraphSpec.AddRestore(spec.RestoreMetadata.ProjectUniqueName);
 
-            var requestProvider = new DependencyGraphSpecRequestProvider(new RestoreCommandProvidersCache(), dependencyGraphSpec);
+            var requestProvider = new DependencyGraphSpecRequestProvider(_providerCache, dependencyGraphSpec);
 
             var restoreArgs = new RestoreArgs
             {
                 AllowNoOp = true,
                 CacheContext = _context,
-                CachingSourceProvider = new CachingSourceProvider(new PackageSourceProvider(_settings)),
-                Log = new NuGetLogger(_logger),
+                CachingSourceProvider = _cachingSourceProvider,
```

Comment:
The caching source provider is supposed to be shared across different restore instances, so caching it is ok.

Reply:
@brandonh-msft you may want to look across our code base and see if we could see a perf increase by using a shared instance of this provider